core-kotlin-modules/core-kotlin-10/src/test/kotlin/com/baeldung/methodAccessInCompanionObject/MethodAccessInCompanionObjectUnitTest.kt

```diff
@@ -0,0 +1,89 @@
+package com.baeldung.methodAccessInCompanionObject
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+
+class MethodAccessInCompanionObjectUnitTest {
+    @Test
+    fun `calls outer method using reference to outer class`() {
+        val result = OuterClass.companionMethod()
+
+        assertEquals("This is a method outside the companion object", result)
+    }
+
+    @Test
+    fun `calls outer method using interface`() {
+        val myClass = OuterClassWIthInterface()
+        val result = OuterClassWIthInterface.companionMethod(myClass)
+
+        assertEquals("This is a method outside the companion object", result)
+    }
+
+    @Test
+    fun `calls outer method using object declaration`() {
+        val result = ClassWithObject.companionMethod()
+
+        assertEquals("This is a method outside the companion object", result)
+    }
+
+    @Test
+    fun `calls outer method using function reference`() {
+        val myClass = ClassWithFunctionReference()
+        val result = ClassWithFunctionReference.companionMethod(myClass::outerClassMethod)
+
+        assertEquals("This is a method outside the companion object", result)
+    }
+}
+
+class OuterClass {
+    companion object {
+        val outerClass: OuterClass = OuterClass()
+        fun companionMethod(): String {
+            return outerClass.outerClassMethod()
+        }
+    }
+
+    fun outerClassMethod(): String {
+        return "This is a method outside the companion object"
+    }
+}
+interface OuterClassInterface {
+    fun outerClassMethod(): String
+}
+
+class OuterClassWIthInterface : OuterClassInterface {
```

Comment:
There's an uppercase 'I' in the "With" part of the name. Lowercase it so the name is OuterClassWithInterface.